floodsub.go

```diff
@@ -309,7 +339,23 @@ func (p *PubSub) handleIncomingRPC(rpc *RPC) error {
 			continue
 		}
 
-		p.maybePublishMessage(rpc.from, pmsg)
+		subs := p.getSubscriptions(&Message{pmsg}) // call before goroutine!
+
+		select {
+		case p.throttleValidate <- struct{}{}:
+			go func(pmsg *pb.Message) {
+				defer func() { <-p.throttleValidate }()
+
+				if p.validate(subs, &Message{pmsg}) {
+					p.sendMsg <- sendReq{
+						from: rpc.from,
+						msg:  &Message{pmsg},
+					}
+				}
+			}(pmsg)
+		default:
+			log.Warning("could not acquire validator; dropping message")
```

Comment:
Maybe in the default here, we can have another select, just like this one, but instead of a default there, we have a timeout. That way we avoid starting a timer for every message, and we don't immediately drop any overflow

Reply:
Wouldn't that block the main loop? And what would be the difference to just using the timeout instead of the default?